src/preferences/effectmanifesttablemodel.cpp

```diff
@@ -0,0 +1,153 @@
+    removeRows(0, m_manifests.size());
+    beginInsertRows(QModelIndex(), 0, newList.size() - 1);
+    m_manifests = newList;
+    endInsertRows();
+}
+
+int EffectManifestTableModel::rowCount(const QModelIndex& parent) const {
+    Q_UNUSED(parent);
+    return m_manifests.count();
+}
+
+int EffectManifestTableModel::columnCount(const QModelIndex& parent) const {
+    Q_UNUSED(parent);
+    return kNumberOfColumns;
+}
+
+QVariant EffectManifestTableModel::headerData(
+        int section, Qt::Orientation orientation, int role) const {
+    if (orientation == Qt::Horizontal) {
+        if (role == Qt::DisplayRole) {
+            if (section == kColumnType) {
+                return tr("Type");
+            } else if (section == kColumnName) {
+                return tr("Name");
+            }
+        }
+    }
+    return QVariant();
+}
+
+QVariant EffectManifestTableModel::data(
+        const QModelIndex& index, int role) const {
+    int rowIndex = index.row();
+    if (!index.isValid() || rowIndex >= m_manifests.size()) {
+        return QVariant();
+    }
+
+    EffectManifestPointer pManifest = m_manifests.at(rowIndex);
+    VERIFY_OR_DEBUG_ASSERT(pManifest) {
+        return QVariant();
+    }
+    int column = index.column();
+    if (column == kColumnName && role == Qt::DisplayRole) {
+        return pManifest->displayName();
+    } else if (column == kColumnType && role == Qt::DisplayRole) {
+        return pManifest->translatedBackendName();
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags EffectManifestTableModel::flags(const QModelIndex& index) const {
+    return QAbstractTableModel::flags(index) | Qt::ItemIsSelectable |
+            Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled;
+}
+
+QMimeData* EffectManifestTableModel::mimeData(
+        const QModelIndexList& indexes) const {
+    QMimeData* mimeData = new QMimeData();
+    QStringList manifestUniqueIds;
+
+    for (const auto& index : indexes) {
+        if (!index.isValid() || index.row() < 0 ||
+                index.row() >= m_manifests.size()
+                // QModelIndexes for both columns are dragged, but only one of them in each row is needed.
+                || index.column() != 0) {
+            continue;
+        }
+        EffectManifestPointer pManifest = m_manifests.at(index.row());
+        manifestUniqueIds << pManifest->uniqueId();
+    }
+    mimeData->setText(manifestUniqueIds.join(kMimeTextDelimiter));
+    return mimeData;
+}
+
+bool EffectManifestTableModel::dropMimeData(const QMimeData* data,
+        Qt::DropAction action,
+        int row,
+        int column,
+        const QModelIndex& parent) {
+    Q_UNUSED(column);
+    if (!data->hasText() || action != Qt::MoveAction) {
+        return false;
+    }
+    QStringList mimeTextLines =
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+            data->text().split(kMimeTextDelimiter, Qt::SkipEmptyParts);
+#else
+            data->text().split(kMimeTextDelimeter, QString::SkipEmptyParts);
```

Comment:
```suggestion
            data->text().split(kMimeTextDelimiter, QString::SkipEmptyParts);
```
typo

Reply:
Fixed in b4c0e78a20e0c8c18352659542ae78db36a9f54c. KDevelop's rename feature didn't rename this because the #if condition wasn't met.